Makefile

```diff
@@ -1,21 +1,25 @@
 .POSIX:
-.PHONY: all compile test clean check
+.PHONY: all compile test clean check install
 .SUFFIXES: .el .elc
 
+DESTDIR ?=
+PREFIR  ?= /usr/local
+DATADIR ?= $(PREFIX)/data
+ELDIR   ?= $(DATADIR)/emacs/site-lisp
+
+
 EMACS = emacs
 MAKEINFO = makeinfo
 BYTEC = compat-25.elc \
 	compat-26.elc \
 	compat-27.elc \
 	compat-28.elc \
 	compat-29.elc \
-	compat.elc \
-	compat-macs.elc \
-	compat-tests.elc
+	compat.elc
 
 all: compile
 
-compile: $(BYTEC)
+compile: $(BYTEC) compat-tests.elc compat-macs.elc
```

Comment:
Yes, that's the better approach.

Reply:
@minad Is there some issue regarding me modifying anything even if the end result is still the same? It does sound like I'm crossing a line just for editing a variable that you wrote.

Reply:
> Alternatively, you can also `filter-out` the files from `BYTEC` you don't want below.

Filter-out would look like this:
`install-compat-el: $(filter-out compat-tests.elc compat-tests.el compat-macs.elc compat-macs.el, $(BYTEC) $(BYTEC:.elc=.el))`

Which means that each file has to be listed twice. Easier would be to separate runtime and development files like I did now.

Reply:
@Thaodan 

> Is there some issue regarding me modifying anything even if the end result is still the same? It does sound like I'm crossing a line just for editing a variable that you wrote.

It is important for development and CI that we maintain the current way the Makefile works. It seems that there was a misunderstandings about what bytecompilation is supposed to achieve here. I hope that I clarified this by adding the additional target force-compile in a417d78f8abfc46386a4008fd694b77ba3e8fb2b. 

I am not happy if I try to explain something and the response is "It doesn't make sense" as in 
https://github.com/emacs-compat/compat/pull/30#discussion_r1338663916.

There is no need to add the install target here. As mentioned before, the recommended way to install the package is via package-install from ELPA. The purpose of the Makefile is foremost to help development and CI. 

I also think the patterns mentioned by @phikal should be explained better. The Makefile should ideally work with different Make implementations, e.g., on BSD. There has been at least one change to the Makefile where BSD support has been restored.

If our opinions about the Makefile clash too much, it may be easier if you simply maintain the few commands in your build script directly. Many ELPA packages don't come with Makefiles so the patterns used in other packages should transfer over.

Reply:
> > Is there some issue regarding me modifying anything even if the end result is still the same? It does sound like I'm crossing a line just for editing a variable that you wrote.
> 
> It is important for development and CI that we maintain the current way the Makefile works. It seems that there was a misunderstandings about what bytecompilation is supposed to achieve here. I hope that I clarified this by adding the additional target force-compile in [a417d78](https://github.com/emacs-compat/compat/commit/a417d78f8abfc46386a4008fd694b77ba3e8fb2b).

The makefile works the same way, I separated the runtime and development dependencies. Doing so is less complicated and doesn't require filter-out which is a GNU make extension.


> I am not happy if I try to explain something and the response is "It doesn't make sense" as in
https://github.com/emacs-compat/compat/pull/30#discussion_r1338663916.

It does or didn't make sense to me, I'm sorry if offended you there, it wasn't my intend.